system-tests/lib/cre/workflow/compile.go

```diff
@@ -13,26 +14,99 @@ import (
 	"github.com/pkg/errors"
 )
 
+type Language = string
+
+const (
+	LanguageGo Language = "go"
+	LanguageTS Language = "typescript"
+)
+
 // CompileWorkflow compiles a workflow from a file path (absolute or relative) and returns the path to the compiled workflow.
 // workflowFilePath is the path to the workflow file.
 // workflowName is the name of the workflow.
 // It will return the path to the compiled workflow.
 // It will return an error if the workflow name is less than 10 characters long.
 // It will return an error if the workflow file path is not a valid file path.
-func CompileWorkflow(workflowFilePath, workflowName string) (string, error) {
+func CompileWorkflow(ctx context.Context, workflowFilePath, workflowName string) (string, error) {
 	if len(workflowName) < 10 {
 		return "", errors.New("workflow name must be at least 10 characters long")
 	}
+
+	language, lErr := delectLanguage(workflowFilePath)
+	if lErr != nil {
+		return "", errors.Wrap(lErr, "failed to detect workflow language")
+	}
+
+	var workflowWasmAbsPath string
+	var err error
+	switch language {
+	case LanguageGo:
+		workflowWasmAbsPath, err = compileGoWorkflow(ctx, workflowFilePath, workflowName)
+	case LanguageTS:
+		workflowWasmAbsPath, err = compileTSWorkflow(ctx, workflowFilePath, workflowName)
+	default:
+		return "", fmt.Errorf("unsupported workflow language: %s", language)
+	}
+
+	if err != nil {
+		return "", fmt.Errorf("failed to compile %s workflow: %w", language, err)
+	}
+
+	compressedWorkflowWasmPath, compressedWorkflowWasmPathErr := compressWorkflow(workflowWasmAbsPath)
+	if compressedWorkflowWasmPathErr != nil {
+		return "", errors.Wrap(compressedWorkflowWasmPathErr, "failed to compress workflow")
+	}
+
+	defer func() {
+		_ = os.Remove(workflowWasmAbsPath)
+	}()
+
+	return compressedWorkflowWasmPath, nil
+}
+
+func delectLanguage(workflowFilePath string) (Language, error) {
+	ext := strings.ToLower(filepath.Ext(workflowFilePath))
+	switch ext {
+	case ".ts", ".tsx":
+		return LanguageTS, nil
+	case ".go":
+		return LanguageGo, nil
+	default:
+		return "", fmt.Errorf("unsupported workflow file extension: %s", ext)
+	}
+}
+
+func compileTSWorkflow(ctx context.Context, workflowFilePath, workflowName string) (string, error) {
+	workflowWasmPath := workflowName + ".wasm"
+
+	buffer := bytes.Buffer{}
+	compileCmd := exec.CommandContext(ctx, "bun", "cre-compile", workflowFilePath, filepath.Join(filepath.Dir(workflowFilePath), workflowWasmPath)) // #nosec G204 -- we control the value of the cmd so the lint/sec error is a false positive
+	compileCmd.Stdout = &buffer
+	compileCmd.Stderr = &buffer
+	if err := compileCmd.Run(); err != nil {
+		fmt.Fprint(os.Stderr, buffer.String())
```

Comment:
Is this different from https://pkg.go.dev/os/exec#Cmd.CombinedOutput?
```suggestion
	if output := compileCmd.CombinedOutput(); err != nil {
		fmt.Fprint(os.Stderr, output)
```